client/controller/shared.js

```diff
@@ -35,25 +37,65 @@ export function makeLayoutMiddleware( LayoutComponent ) {
 	};
 }
 
+function loadCSS( cssUrl, elementId = null ) {
+	if ( typeof document === 'undefined' ) {
+		return;
+	}
+
+	let link;
+	if ( elementId ) {
+		link = document.getElementById( elementId );
+	}
+
+	if ( link ) {
+		if ( link.getAttribute( 'href' ) !== cssUrl ) {
+			link.href = cssUrl;
+		}
+		return;
+	}
+
+	link = Object.assign( document.createElement( 'link' ), {
+		rel: 'stylesheet',
+		type: 'text/css',
+		href: cssUrl,
+		... elementId && { id: elementId }
+	} );
+
+	document.head.appendChild( link );
+}
+
 export function setSection( section ) {
 	return ( context, next = noop ) => {
 		context.store.dispatch( setSectionAction( section ) );
 
-		next();
+		loadSectionCSS( context, next );
```

Comment:
It should be `next()` IMO.

Consider following scenario:
User is using RTL locale, first entrance here, `context.isRTL` is `undefined`,
`loadCSS( 'login.css' )` is called,
then, we enter `setUpLocale`, now `context.isRTL` is set, so we call `loadCSS( 'login.rtl.css' )`...

Reply:
The discussion got lost on a different commit:

https://github.com/Automattic/wp-calypso/pull/17988/files/252109f4465768a9800e1c8bdc8bbda00b9ab410..2ac676f1c435e946b94a63aa1956df1fd88b7ef6#r141019050